### PR TITLE
TD: add cat_cafe_get_thread_cats MCP tool + de-hardcode catId (#408)

### LINKED ISSUE #408: TD: de-hardcode catId in MCP tool descriptions + add cat_cafe_get_thread_cats
## Background

MCP tool descriptions hardcode static catId examples like \`["opus", "codex", "gemini"]\` in multiple \`.describe()\` calls. While the runtime validation uses dynamic \`CatRegistry\`, the schema hints mislead LLMs into thinking only those specific catIds are valid.

Additionally, there is no MCP tool for cats to discover which cats are in the current thread. The API already has \`GET /api/threads/:id/cats\` but no callback equivalent for MCP use.

## Scope

1. **De-hardcode catId examples** in MCP tool descriptions:
   - \`callback-tools.ts\`: multiMention, startVote, updateBootcampState, updateWorkflow, postMessage targetCats
   - \`session-chain-tools.ts\`: listSessionChain catId filter

2. **Add \`cat_cafe_get_thread_cats\` MCP tool**:
   - New callback endpoint: \`GET /api/callbacks/thread-cats\`
   - Auth: invocationId + callbackToken (standard callback pattern)
   - **Current-thread-only** — resolves threadId from invocation record, no override parameter (principle of least privilege; prevents cross-thread snooping via MCP callbacks)
   - Returns: \`threadId\`, \`participants\`, \`routableNow\`, \`routableNotJoined\`, \`notRoutable\`, \`routingPolicy\`

## Non-goals

- No global \`list_members\` tool with \`online/offline/busy/model/capabilities\` (no stable truth source for those yet)
- No dynamic injection of cat names into all tool descriptions (overkill)
- No cross-thread threadId override (least privilege — cats should only discover members in their own thread)

## Review criteria (codex)

- No second member status model invented
- No bypass of existing auth / current-thread resolution
- No hardcoded catId examples remaining in schema descriptions
- Test coverage for default-thread scenario and auth edge cases

[宪宪/Opus-46🐾]

### PR DESCRIPTION
**Pull request by mindfn**
## Summary

- Add `cat_cafe_get_thread_cats` MCP tool for cats to discover thread participants and routable cats
- Extract shared `categorizeThreadCats()` to eliminate logic duplication between F142 route and new callback
- Clean up 7 hardcoded catId examples (`opus/codex/gemini`) in MCP tool descriptions, replacing with `get_thread_cats` references

## Changes

| File | What |
|------|------|
| `packages/api/src/routes/thread-cats-core.ts` | **New** — shared categorization logic (single truth source) |
| `packages/api/src/routes/callback-thread-cats-routes.ts` | **New** — `GET /api/callbacks/thread-cats` callback endpoint |
| `packages/api/src/routes/thread-cats.ts` | Refactored to use shared core |
| `packages/api/src/routes/callbacks.ts` | Register new route + `agentRegistry` dep |
| `packages/api/src/index.ts` | Pass `agentRegistry` to callback opts |
| `packages/mcp-server/src/tools/callback-tools.ts` | New tool + description cleanup |
| `packages/mcp-server/src/tools/session-chain-tools.ts` | Description cleanup |
| `packages/mcp-server/src/tools/index.ts` | Export new tool |
| `packages/api/test/callback-thread-cats.test.js` | **New** — 5 tests (auth, 404, categorization) |
| `packages/mcp-server/test/tool-registration.test.js` | Registration guard updated |

## Response structure

`cat_cafe_get_thread_cats` returns (current-thread-only, resolved from invocation record):

```json
{
  "threadId": "thread_xxx",
  "participants": [...],
  "routableNow": [...],
  "routableNotJoined": [...],
  "notRoutable": [...],
  "routingPolicy": "v2" | null
}
```

## Review

Reviewed by codex (缅因猫) — 2 rounds:
- R1: P2 (logic duplication) + P3 (missing categorization assertions) → fixed in `fb7feb06`
- R2: All 4 checkpoints pass, no new issues

Maintainer review by zts212653:
- P2 (scope alignment): Issue #408 updated to current-thread-only (least privilege)
- P3 (`.extend({})` no-op): Removed in `56594ca`

## Test plan

- [x] `callback-thread-cats.test.js` — 5/5 pass (400/401/404/200+categorization/empty-threadId)
- [x] `thread-cats.test.js` — 7/7 pass (existing F142 tests, no regression)
- [x] `tool-registration.test.js` — 10/10 pass (new tool in expected lists)
- [x] TypeScript type check — clean
- [x] Biome lint — clean

Closes #408

🐾 Generated with [Claude Code](https://claude.com/claude-code) [宪宪/Opus-46🐾]

### CODE REVIEW

**Comment by mindfn** [comment]
@codex review

**Comment by mindfn** [comment]
@codex review

Please review latest commit 4eda0edd for P1/P2 only.

规则：任何 P1/P2 必须给"可执行复现"：
- 优先：新增/更新一个 failing test（最小复现）
- 否则：给确定性复现步骤（命令 + 输入 + 预期/实际）
没有证据的一律降级为 P3 建议，不算缺陷。

审查标准（详见 AGENTS.md "Review guidelines" section）：
- P0 数据丢失/安全漏洞 | P1 逻辑错误/测试缺失/架构违规
- P2 性能/重复/命名 | P3 风格偏好
- 禁止 `any`、文件 200 行警告/350 硬上限、新功能必须有测试

**Review by zts212653 — COMMENTED** [review]
## Maintainer Inbound Review — 布偶猫/宪宪 (Opus)

**Direction: WELCOME** — this fills a real gap. Cats currently have no MCP-accessible way to discover valid catIds before calling `multi_mention` / `start_vote` / routing tools. The DRY extraction of `categorizeThreadCats()` and the de-hardcoding of static catId examples are both strictly beneficial.

### What works well

- **Callback auth pattern** follows existing conventions exactly (invocationId + callbackToken). Clean.
- **`categorizeThreadCats()` as a pure function with DI** — no imports from core modules, fully testable, single source of truth for KD-9 categorization. This is exactly how we'd do it internally.
- **Test coverage** is solid: 5 tests covering 400/401/404/200/empty-threadId edge cases.
- **Non-goals** are well-scoped: no premature member status model, no dynamic schema injection.

### Issues to address

**P2 — Spec/implementation scope mismatch**

Issue #408 states: *"Default to current invocation's threadId, optional override"* and review criteria include *"default-thread and cross-thread scenarios."*

The implementation is current-thread-only: `getThreadCatsInputSchema = {}` (empty), route reads `record.threadId` with no override param, tests only cover the default-thread path.

I believe current-thread-only is actually the **better design** — it prevents cross-thread snooping via MCP callbacks and follows the principle of least privilege. But the issue description and the implementation need to agree. Please either:
- **(Preferred)** Update #408 scope to explicitly state current-thread-only, remove the "optional override" language and "cross-thread" review criterion, OR
- Add the optional threadId override with proper cross-thread auth checks

**P3 — Minor cleanup**

1. `threadCatsQuerySchema = callbackAuthSchema.extend({})` — the `.extend({})` is a no-op. Just use `callbackAuthSchema` directly.

2. `ParticipantActivityInput` drops `lastResponseHealthy?: boolean` from the original interface. This is functionally safe (categorization doesn't use it, API response doesn't include it), but worth noting for forward-compat if we ever want health-aware categorization in the shared core.

3. Multi-line JSDoc blocks and `TD #408` references in comments — our downstream style is minimal-comments. Not blocking, but we'll strip these during intake.

### Process note

Issue #408 currently has no labels. For our inbound flow, it needs at least `triaged` + a type label before merge. Happy to add these ourselves as maintainers if the above P2 is resolved.

---

Overall: close to merge-ready. The P2 scope alignment is the one thing to resolve — the code quality itself is solid work. 🐾

File: packages/api/src/routes/callback-thread-cats-routes.ts [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+import { callbackAuthSchema } from './callback-auth-schema.js';
+import { EXPIRED_CREDENTIALS_ERROR } from './callback-errors.js';
+import { categorizeThreadCats } from './thread-cats-core.js';
+
+interface ThreadCatsCallbackDeps {
+  registry: InvocationRegistry;
+  threadStore: IThreadStore;
+  agentRegistry: { getAllEntries(): Map<string, unknown> };
+}
+
```
Comment (zts212653) [inline packages/api/src/routes/callback-thread-cats-routes.ts:23]: `callbackAuthSchema.extend({})` — the `.extend({})` is a no-op, just use `callbackAuthSchema` directly. (P3)

File: packages/mcp-server/src/tools/callback-tools.ts [hunk 2]
```diff
@@ -714,6 +714,14 @@ export async function handleBootcampEnvCheck(input: { threadId: string }): Promi
   return callbackPost('/api/callbacks/bootcamp-env-check', { threadId: input.threadId });
 }
 
+// ============ Thread Cats Discovery (TD #408) ============
+
+export const getThreadCatsInputSchema = {};
+
```
Comment (zts212653) [inline packages/mcp-server/src/tools/callback-tools.ts:720]: This empty input schema is the core of the P2 scope gap: issue #408 specifies "optional override" for threadId, but this tool takes no input at all. Current-thread-only is the safer design IMO — just needs the issue description updated to match.

**Comment by mindfn** [comment]
## Review 修复确认

| # | 问题 | 状态 | 处理方式 |
|---|------|------|----------|
| P2 | Spec/implementation scope mismatch | ✅ | Issue #408 scope 已更新：明确 current-thread-only 设计（least privilege），移除 "optional override" 和 cross-thread 语言 |
| P3.1 | `.extend({})` no-op | ✅ | 直接使用 `callbackAuthSchema`，去掉空 extend |
| P3.2 | `ParticipantActivityInput` 丢了 `lastResponseHealthy` | Noted | 有意为之——shared core 按 DI 设计，categorization 不需要 health 信息。未来需要 health-aware routing 时扩展 input 接口 |
| P3.3 | JSDoc 风格 | Noted | 不阻塞，downstream intake 时处理 |

**Process note**: Issue #408 labels — 请 maintainer 添加 `triaged` + type label。

测试：callback-thread-cats 5/5 passed, biome clean, lint clean
Commit: `4c3df1cc` — fix(#408): align scope + remove .extend({}) no-op (review P2/P3)

[宪宪/Opus-46🐾]

File: packages/api/src/routes/callback-thread-cats-routes.ts (hunk 1 above)
Comment (mindfn) [reply packages/api/src/routes/callback-thread-cats-routes.ts:23]: Fixed in 4c3df1cc — removed `.extend({})`, now uses `callbackAuthSchema` directly. [宪宪/Opus-46🐾]

File: packages/mcp-server/src/tools/callback-tools.ts (hunk 2 above)
Comment (mindfn) [reply packages/mcp-server/src/tools/callback-tools.ts:720]: Agreed — current-thread-only is the better design. Issue #408 scope updated to explicitly state this: no override parameter, least privilege, prevents cross-thread snooping via MCP callbacks. [宪宪/Opus-46🐾]
